pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -94,8 +94,15 @@ func (pl *NodeAffinity) PreFilter(ctx context.Context, cycleState *framework.Cyc
 	affinity := pod.Spec.Affinity
 	if affinity == nil ||
 		affinity.NodeAffinity == nil ||
-		affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil ||
-		len(affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms) == 0 {
```

Comment:
shouldn't we preserve this line here?

So that if there is no `RequiredDuringSchedulingIgnoredDuringExecution` and no `addedNodeSelector` and no `.Spec.NodeSelector`, we also return Skip.

Reply:
You mean we can add `len(affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms) == 0` here, right? I think that's true. Fixed

Reply:
Ohh, that's true. Let's keep the behavior.

Btw, I checked the validation logic, and Pods with a nil []NodeSelectorTerm should be rejected actually. 
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L4012-L4014

Reply:
ahh, good find. It should be fine to remove the test case. Up to you.